src/components/NewCommentForm.vue

```diff
@@ -0,0 +1,156 @@
+<script lang="ts">
+import type { PropType } from 'vue'
+import { useUserStore } from '@/stores/users'
+import InputTemplate from '../ui/inputTemplate.vue'
+import ButtonTemplate from '../ui/buttonTemplate.vue'
+import { usePostsStore } from '@/stores/posts'
+import { postNewComment } from '@/api/comments'
+import type { CustomComment } from '@/types/CustomComment'
+import type { NewComment } from '@/types/Comment'
+import type { Post } from '@/types/Post'
+
+export default {
+  components: { InputTemplate, ButtonTemplate },
+
+  props: {
+    selectedPost: Object as PropType<Post | null>
+  },
+
+  watch: {
+    selectedPost() {
+      if (this.selectedPost) {
+        this.isNameError = false
+        this.isEmailError = false
+        this.isTextError = false
+        this.commentText = ''
+      }
+    }
+  },
+
+  data() {
+    return {
+      isNameError: false,
+      isEmailError: false,
+      isTextError: false,
+      submitting: false,
+      authorName: '',
+      authorEmail: '',
+      commentText: '',
+      newCommentError: false
+    }
+  },
+
+  emits: ['addNewComment'],
+
+  methods: {
+    handleSubmit() {
+      this.submitting = true
+      this.isEmailError = !this.authorEmail
+      this.isNameError = !this.authorName
+      this.isTextError = !this.commentText
+
+      if (!this.authorEmail || !this.authorName || !this.commentText) {
+        this.submitting = false
+        return
+      }
+      const userStore = useUserStore()
+
+      if (this.selectedPost && userStore.user.id) {
+        const newComment: NewComment = {
+          postId: this.selectedPost.id,
+          name: this.authorName,
+          email: this.authorEmail,
+          body: this.commentText
+        }
+
+        postNewComment(newComment)
+          .then(({ data }) => {
+            this.$emit('addNewComment', data)
+          })
+          .catch(() => (this.newCommentError = true))
+          .finally(() => this.resetForm())
+      } else {
+        const postsList = usePostsStore()
+        const post = postsList.posts.find((post) => post.id === this.selectedPost?.id)
+
+        if (post) {
+          const id = post.comments[post.comments.length - 1]?.id + 1 || 1
```

Comment:
You can describe generateId func as helper, than use it in this and others place

Reply:
Super! Thanks!)